server/src/main/java/org/opensearch/plugins/NoOpResourceAccessControlPlugin.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+import org.opensearch.accesscontrol.resources.EntityType;
+import org.opensearch.accesscontrol.resources.ResourceSharing;
+import org.opensearch.accesscontrol.resources.ShareWith;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This plugin class defines a no-op implementation of Resource Plugin.
+ *
+ * @opensearch.experimental
+ */
+public class NoOpResourceAccessControlPlugin implements ResourceAccessControlPlugin {
+
+    /**
+     *
+     * @return an empty map of resource names accessible by this user.
+     */
+    @Override
+    public Map<String, List<String>> listAccessibleResources() {
+        return Map.of();
```

Comment:
In NoOp implementation, wouldn't a plugin expect to be able to list all? Why the return type `Map<String, List<String>>`? Can this be made more generic and let the plugin that defines the Resource to be able to serialize and deserialize what is stored in a resource index?